File: tests/dart/field_access/test/primitives.dart
import 'package:test/test.dart';

import '../lib/generated/rid_api.dart';

void main() {
  test('field_access: primitives', () {
    final store = Store.instance;
    // unsigned ints
    expect(store.sId, 1, reason: 'u8');
    expect(store.mId, 10, reason: 'u16');
    expect(store.lId, 100, reason: 'u32');
    expect(store.xlId, 1000, reason: 'u64');

    // signed ints
    expect(store.sSigned, -1, reason: 'i8');
    expect(store.mSigned, -10, reason: 'i16');
    expect(store.lSigned, -100, reason: 'i32');
    expect(store.xlSigned, -1000, reason: 'i64');

    // bool
    expect(store.ok, true, reason: 'bool: true');
    expect(store.notOk, false, reason: 'bool: false');
  });
}

File: tests/dart/field_access/src/primitives.rs
use rid::RidStore;

#[rid::store]
pub struct Store {
    // unsigned integers
    s_id: u8,
    m_id: u16,
    l_id: u32,
    xl_id: u64,
    // signed integers
    s_signed: i8,
    m_signed: i16,
    l_signed: i32,
    xl_signed: i64,
    // bool
    ok: bool,
    not_ok: bool,
}

impl RidStore<Msg> for Store {
    fn create() -> Self {
        Self {
            s_id: 1,
            m_id: 10,
            l_id: 100,
            xl_id: 1000,

            s_signed: -1,
            m_signed: -10,
            l_signed: -100,
            xl_signed: -1000,

            ok: true,
            not_ok: false,
        }
    }

    fn update(&mut self, _req_id: u64, _msg: Msg) {
        unimplemented!()
    }
}

#[rid::message(Reply)]
pub enum Msg {
    NotUsed,
}
#[rid::reply]
pub enum Reply {
    NotUsed,
}
